pandas/_libs/src/ujson/lib/ultrajsondec.c

```diff
@@ -1174,14 +1174,20 @@ JSOBJ JSON_DecodeObject(JSONObjectDecoder *dec, const char *buffer,
     ds.dec = dec;
 
     locale = setlocale(LC_NUMERIC, NULL);
+    if (!locale) {
```

Comment:
Not sure when / how this would happen but our current implementation doesn't check for a successful setlocale call. If that ends up being NULL we have undefined behavior in the subsequent strcmp